### PR TITLE
Uni v4 keyboard add

### PR DESCRIPTION
**Pull request by petercpark**
## Description

Adding another keyboard version. [The Uni v4](https://stenokeyboards.com/products/the-uni-v4) uses the rp2040 chip.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/the_uni/rp_2040/config.h [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+// #define LOCKING_SUPPORT_ENABLE
+
+// /* Locking resynchronize hack */
+// #define LOCKING_RESYNC_ENABLE
+
+// /* key combination for command */
+#define IS_COMMAND() ( \
+    keyboard_report->mods == (MOD_BIT(KC_LSHIFT) | MOD_BIT(KC_RSHIFT)) \
+)
+
```
Comment (zvecr) [inline keyboards/the_uni/rp_2040/config.h:62]:
```suggestion
```

File: keyboards/the_uni/rp_2040/rules.mk [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# MCU name
+MCU = RP2040
+BOOTLOADER = rp2040
+
+ALLOW_WARNINGS = yes
+PICO_INTRINSICS_ENABLED = no # ATM Unsupported by ChibiOS!
+
```
Comment (zvecr) [inline keyboards/the_uni/rp_2040/rules.mk:7]:
```suggestion

```

File: keyboards/the_uni/rp_2040/rp_2040.c [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include QMK_KEYBOARD_H
```
Comment (zvecr) [inline keyboards/the_uni/rp_2040/rp_2040.c:17]:
```suggestion
```

File: keyboards/the_uni/rp_2040/config.h [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x9000
+#define PRODUCT_ID      0x0001
+#define DEVICE_VER      0x0004
+#define MANUFACTURER    StenoKeyboards
+#define PRODUCT         The Uni
+
```
Comment (zvecr) [inline keyboards/the_uni/rp_2040/config.h:27]: Should be configured like `keyboards/the_uni/usb_c/info.json`
```suggestion

```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/the_uni/rp_2040/readme.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
+# Uni v4
+
+![Uni v4](https://cdn.shopify.com/s/files/1/0581/6798/4289/products/top4_1680x.png)
```
Comment (drashna) [inline keyboards/the_uni/rp_2040/readme.md:3]: please rehost this image on Imgur

File: keyboards/the_uni/rp_2040/readme.md [hunk 6]
```diff
@@ -0,0 +1,25 @@
+# Uni v4
+
+![Uni v4](https://i.imgur.com/HyvMu26.png?1)
```
Comment (drashna) [inline keyboards/the_uni/rp_2040/readme.md:3]:
```suggestion
![Uni v4](https://i.imgur.com/HyvMu26h.png)
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/the_uni/pro_micro/readme.md [hunk 7]
```diff
@@ -1,7 +1,9 @@
-# The Uni
+# The Uni v1 and v2
+
+![Uni v2](https://github.com/petercpark/The_Uni/blob/main/Pics/uni%20v2/uni-v2.JPG?raw=true)
```
Comment (drashna) [inline keyboards/the_uni/pro_micro/readme.md:3]: Please rehost on Imgur.

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/the_uni/rp_2040/rules.mk [hunk 8]
```diff
@@ -0,0 +1,16 @@
+# MCU name
+MCU = RP2040
+BOOTLOADER = rp2040
```
Comment (fauxpark) [inline keyboards/the_uni/rp_2040/rules.mk:3]:
```suggestion

# Bootloader selection
BOOTLOADER = rp2040

```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Pull request merged**